sisl/viz/plotly/plots/grid.py

```diff
@@ -512,6 +512,8 @@ def _set_data(self, axes, nsc, interp, trace_name, transforms, represent, cut_va
         elif represent.endswith("phase"):
             values = np.angle(grid.grid, deg=represent.startswith("deg"))
 
+        grid.grid = values
+
         #Remove the leftover dimensions
         values = np.squeeze(values)
 
```

Comment:
what about using `plot_func = getattr(self, f"_plot{values.ndim}D")` ?
That would also raise an issue for erroneous data?

Reply:
Done! (not pushed yet)